server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/InternalDateHistogram.java

```diff
@@ -394,9 +395,47 @@ protected Bucket reduceBucket(List<Bucket> buckets, ReduceContext context) {
         return createBucket(buckets.get(0).key, docCount, aggs);
     }
 
-    private void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
+    private int getTotalBucketCount() {
+        LongBounds bounds = emptyBucketInfo.bounds;
+        int bucketCount = 0;
+        if (bounds != null && bounds.getMin() != null && bounds.getMax() != null) {
+            long min = bounds.getMin() + offset;
+            long max = bounds.getMax() + offset;
+            long intervalWidth = 0;
+            int i = 0;
+            long key = min;
+            while (key < max && i++ < 10) {
+                bucketCount++;
+                long nextKey = nextKey(key).longValue();
+                intervalWidth = Math.max(intervalWidth, nextKey - key);
+                key = nextKey;
+            }
+            if (bucketCount < 10) {
+                return bucketCount;
+            }
+            long estimatedBuckets = Math.round(Math.ceil((double) (max - min) / intervalWidth));
+            if (estimatedBuckets > Integer.MAX_VALUE) {
+                return Integer.MAX_VALUE;
+            }
+            return (int) estimatedBuckets;
+        }
+        return 0;
+    }
+
+    protected void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
         Bucket lastBucket = null;
         LongBounds bounds = emptyBucketInfo.bounds;
+        int preEmptyBucketCount = list.size();
+        int emptyBucketCount = getTotalBucketCount() - list.size();
+        if (emptyBucketCount > 0) {
+            CircuitBreaker breaker = reduceContext.getBreaker();
+            if (breaker != null) {
+                breaker.addEstimateBytesAndMaybeBreak(50L * emptyBucketCount, "empty date histogram buckets");
+            }
+            preEmptyBucketCount += emptyBucketCount;
```

Comment:
preEmptyBucketCount basically is getTotalBucketCount, can remove and just use that

Reply:
Yeah, it was getting a little confusing. I simplified it by removing preEmptyBucketCount

Reply:
Spoke too soon, we do need to have the counts before and after. I added a comment to help understand